qiskit_nature/transformers/second_quantization/electronic/seniority_zero_transformer.py

```diff
@@ -0,0 +1,319 @@
+    operator we can use the :class:`~qiskit_nature.mappers.second_quantization.DirectMapper`.
+
+    For more details on the restricted approximation refer to https://arxiv.org/abs/2002.00035.
+
+    Example:
+        The hydrogen molecule in STO-3G basis, which is encoded on four qubits in the standard HF method,
+        while it is encoded on two qubits in the restricted HF formalism.
+
+        .. code-block:: python
+
+            from qiskit_nature.converters.second_quantization import QubitConverter
+            from qiskit_nature.drivers import Molecule
+            from qiskit_nature.drivers.second_quantization import (
+                ElectronicStructureDriverType,
+                ElectronicStructureMoleculeDriver,
+            )
+            from qiskit_nature.mappers.second_quantization import DirectMapper
+            from qiskit_nature.problems.second_quantization import ElectronicStructureProblem
+            from qiskit_nature.transformers.second_quantization.electronic import SeniorityZeroTransformer
+
+            molecule = Molecule(geometry=[["H", [0.0, 0.0, 0.0]], ["H", [0.0, 0.0, 0.8]]], charge=0, multiplicity=1)
+            driver = ElectronicStructureMoleculeDriver(molecule, basis="sto3g", driver_type=ElectronicStructureDriverType.PYSCF)
+
+            problem = ElectronicStructureProblem(driver, transformers=[SeniorityZeroTransformer()])
+            converter = QubitConverter(DirectMapper())
+
+            qubit_op = converter.convert(problem.second_q_ops()[0])
+    """
+
+    def __init__(self):
+        self._num_modes: int = None
+
+    def transform(
+        self, grouped_property: GroupedSecondQuantizedProperty
+    ) -> GroupedElectronicProperty:
+        """Reduces the given `GroupedElectronicProperty` to the restricted Hartree-Fock space.
+
+        Args:
+            grouped_property: the `GroupedElectronicProperty` to be transformed.
+
+        Returns:
+            A new `GroupedElectronicProperty` instance.
+
+        Raises:
+            QiskitNatureError: If the provided `GroupedElectronicProperty` does not contain a
+                               `ElectronicEnergy` or a `ParticleNumber` instance, or if the
+                               alpha and beta integrals stored in the `ElectronicEnergy` are
+                               not identical.
+        """
+        if not isinstance(grouped_property, GroupedElectronicProperty):
+            raise QiskitNatureError(
+                "Only `GroupedElectronicProperty` objects can be transformed by this Transformer, "
+                f"not objects of type, {type(grouped_property)}."
+            )
+
+        electronic_energy = grouped_property.get_property(ElectronicEnergy)
+        if electronic_energy is None:
+            raise QiskitNatureError(
+                "The provided `GroupedElectronicProperty` does not contain a `ElectronicEnergy` "
+                "property, which is required by this transformer!"
+            )
```

Comment:
I think erroring is the correct way to go. Especially with the seniority zero case, if we were to only log a warning/error and try to proceed without the transform being the applied, the problem would immediately be twice the size and a user may not notice and wait a significantly long time to obtain a result (if at all).

Reply:
Yes, in this case, it seems the right thing to do as also the qubit mapper, that was presumably chosen on the assumption this would be in effect, needs to match as well.